### PR TITLE
[RHCLOUD-21260] Add admin_only parameter support for groups/:UID/principals 

### PR DESCRIPTION
**Pull request by lpichler**
## Depends on
- [x] https://github.com/RedHatInsights/backoffice-proxy/pull/108

## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-21260

## Description of Intent of Change(s)
This adds support for admin_only parameter for API endpoint:

```
api/rbac/v1/groups/:UID/principals
```

This was not possible before because this request depends on backoffice proxy. 
It was not possible because RBAC uses call to BOP to validate principals in this request. 

This validation did not support **admin_only** parameter - so support for **admin_only** parameter needed to be added in BOP: https://github.com/RedHatInsights/backoffice-proxy/pull/108

## Local Testing
IF https://github.com/RedHatInsights/backoffice-proxy/pull/108 is not merged we need to run backoffice proxy locally with this PR.

- run rbac
- call request:
preparation:

Prepare data so such request will return you org_admin and non admins:
```
api/rbac/v1/groups/adaad94e-d888-4469-8ed7-160d53f88159/principals/
```

```
"data": [
    {
      "username": "lpichler99",
      "email": "lpichler@redhat.com",
      "first_name": "Libor",
      "last_name": null,
      "is_active": true,
      "is_org_admin": true
    },
    {
      "username": "lpichler98a",
      "email": "lpichler@redhat.com",
      "first_name": "Og",
      "last_name": null,
      "is_active": true,
      "is_org_admin": false
    }
  ]
}
```

With this change when you call 

```
api/rbac/v1/groups/adaad94e-d888-4469-8ed7-160d53f88159/principals/?admin_only=true
```

response should be:
```
"data": [
    {
      "username": "lpichler99",
      "email": "lpichler@redhat.com",
      "first_name": "Libor",
      "last_name": null,
      "is_active": true,
      "is_org_admin": true
    }
  ]
}
```

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [ ] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  - [ ] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by petracihalova — APPROVED** [review]

**Comment by vbelchio** [comment]
LGTM
All the tests are passing locally once we already have the changes in Stage BOP.

**Review by abaiken — APPROVED** [review]
LGTM!

File: docs/source/specs/openapi.json [hunk 1]
```diff
@@ -1186,6 +1186,20 @@
               "format": "uuid"
             }
           },
+          {
+            "name": "admin_only",
+            "in": "query",
+            "description": "Get only admin users within an account.",
+            "required": false,
+            "schema": {
+              "type": "string",
```
Comment (gwenneg) [inline docs/source/specs/openapi.json:1195]: We're about to use this in Notifications. Is there a reason why this field is not a `boolean`?

Comment (petracihalova) [reply docs/source/specs/openapi.json:1195]: hi @gwenneg I believe it is boolean [now](https://github.com/RedHatInsights/insights-rbac/blob/ca5763f67a768aa5d88f73661004820f578dcc72/docs/source/specs/openapi.json#L1211)

Comment (gwenneg) [reply docs/source/specs/openapi.json:1195]: Nice. I should have checked the latest code 😅 Thanks for your help!

Comment (petracihalova) [reply docs/source/specs/openapi.json:1195]: yw, let me know if anything ❤️

**Pull request merged**